tests/inc/Analytics/EnvironmentConfigTest.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types = 1);
+
+namespace RemoteDataBlocks\Tests\Analytics;
+
+use PHPUnit\Framework\TestCase;
+use RemoteDataBlocks\Analytics\EnvironmentConfig;
+
+class EnvironmentConfigTest extends TestCase {
+	public function setUp(): void {
+		$GLOBALS['__wordpress_filters'] = [];
+	}
+
+	public function testIsEnabledViaFilterReturnsFalse(): void {
+		$obj = new EnvironmentConfig();
+
+		$this->assertEquals( false, $obj->is_enabled_via_filter() );
+	}
+
+	public function testIsEnabledViaFilterReturnsTrue(): void {
+		$GLOBALS['__wordpress_filters']['remote_data_blocks_enable_tracks_analytics'] = true;
+		$obj = new EnvironmentConfig();
+
+		$this->assertEquals( true, $obj->is_enabled_via_filter() );
+	}
+
+	public function testIsWpvipSiteReturnsFalse(): void {
+		$obj = new EnvironmentConfig();
+
+		$this->assertEquals( false, $obj->is_wpvip_site() );
+	}
+
+	public function testIsWpvipSiteReturnsTrue(): void {
+		$obj = new EnvironmentConfig();
+		set_private_property( EnvironmentConfig::class, $obj, 'tracks_core_props', [ 'hosting_provider' => 'wpvip' ] );
```

Comment:
Do we absolutely need these private property methods?

Can we mock/stub out `get_tracks_core_properties` instead of setting `tracks_core_props` directly?

Reply:
We're assigning `track_core_props` inside the `init` action hook, so using `set_private_property` seems okay given that we're not running tests within the WP environment.